src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -325,6 +325,175 @@ std::optional<Values> Visitor::visit(Parser::ValuesClauseContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Load is not supported.");
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::ClearContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Clear is not supported.");
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::DropContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Drop is not supported.");
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::CreateContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Create is not supported.");
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::AddContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Add is not supported.");
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::MoveContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Move is not supported.");
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::CopyContext* ctx) {
+  reportNotSupported(ctx, "SPARQL 1.1 Update Copy is not supported.");
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::InsertDataContext* ctx) {
+  return visit(ctx->quadData());
+}
+
+// ____________________________________________________________________________________
+vector<SparqlTripleSimple> Visitor::visit(Parser::DeleteDataContext* ctx) {
+  return visit(ctx->quadData());
+}
+
+// ____________________________________________________________________________________
+void Visitor::visit(Parser::DeleteWhereContext* ctx) {
+  // DeleteData only deletes the specified triples. DeleteWhere matches the
+  // given pattern and then deletes the matched triples. This is a shortcut for
+  // a modify query.
+  reportNotSupported(ctx, "SPARQL 1.1 Update DeleteWhere is not supported.");
+}
+
+// ____________________________________________________________________________________
+ParsedQuery Visitor::visit(Parser::ModifyContext* ctx) {
+  if (ctx->iri()) {
+    // Could also be default; disallow completely for now.
+    reportNotSupported(ctx->iri(), "Named graphs are not supported.");
+  }
+  if (!ctx->usingClause().empty()) {
+    reportNotSupported(ctx->usingClause(0),
+                       "Using is not supported in Update/Insert.");
+  }
+
+  parsedQuery_._rootGraphPattern = visit(ctx->groupGraphPattern());
+
+  parsedQuery_._clause = parsedQuery::UpdateClause();
+  if (ctx->deleteClause()) {
+    parsedQuery_.updateClause().toDelete_ = visit(ctx->deleteClause());
+  }
+  if (ctx->insertClause()) {
+    parsedQuery_.updateClause().toInsert_ = visit(ctx->insertClause());
+  }
```

Comment:
I've seen this before, factr it out e.g.

`parsedQuery_._clause = visitDeleteAndInsert(ctx->deleteClause(), ctx->insertClause())`
You can then even use this to implement the syntactic sugar shorthands like insertWhere and deleteWhere, maybe you need another helper that also directly handles the `rootGraphPattern`.